### ISSUE TITLE
EventRecord indexing mismatch (latest master)

### ISSUE THREAD

**Issue by jacogr** [paritytech/substrate#2021]
There seems to be some mismatch in the `Vec<EventRecord>` via a `state_getStorage` query. After submitting a transfer on latest master, having it succeed and making a call to get the storage entry at that specific block -

`{"id":14,"jsonrpc":"2.0","method":"state_getStorage","params":["0xcc956bdb7605e3547539f321ac2bc95c","0x5390b55c8443239c61558df7a1bdeacbf9d89df30192938e2b9d2178e8087d79"]}`

This is the resulting JSON that is being returned -

`{"jsonrpc":"2.0","result":"0x140000000000000000010000000203306721211d5404bd9da88e0204360a1a9ab8b87c66c1bc2fcdd37f3c2222cc204a83f1c09be797bc3d9adce29818368b276a84e6b545ced492c25c948978d7f83930000000000000000000000000000000000000000000000000000000000000000100000000000103006400000000000000010d000100000001000000000000000000000000000000","id":14}`

If we manually decode it -

```
14 // length 5 (20 >> 2 = 5)

// *** 0
00 // ApplyExtrinsic
00000000 // 0(u32)
0000 // system, ExtrinsicSuccess

// **** 1
00 // ApplyExtrinsic
01000000 // 1(u32)
0203 // balances, ??? there are only 3 in total, this refers to the 4th

306721211d5404bd9da88e0204360a1a9ab8b87c66c1bc2fcdd37f3c2222cc204a83f1c09be797bc3d9adce29818368b276a84e6b545ced492c25c948978d7f83930000000000000000000000000000000000000000000000000000000000000000100000000000103006400000000000000010d000100000001000000000000000000000000000000
```

So assuming `0203` should have been `0202`, i.e the balances only have 3 events currently defined in the srml, where `02` is `Transfer`.

**Comment by gavofyork** [comment]
@thiolliere any idea about what could cause that?

**Comment by thiolliere** [comment]
oh indeed yes, because RawEvent has a new (doc hideen) field for constraining its instance, https://github.com/paritytech/substrate/blob/91fa6594fe734ae41a2019cbc277cd175e7e91b9/srml/support/src/event.rs#L284-L290

This is a breakage indeed, I'll fix it by making a PR that add this field at the end of all other instead of as first field

**Comment by bkchr** [comment]
Fixed by #2030
